tests/unit/type/set.tcl

```diff
@@ -1168,6 +1168,111 @@ foreach type {single multiple single_multiple} {
     }
 }
 
+start_server {
+    tags {"set external:skip"}
+    overrides {
+        "set-max-intset-entries" 1
+        "allow-empty-set" "yes"
+    }
+} {
+    proc save_load_rdb {{config_override {}}} {
+        r save
+        restart_server 0 true false true now $config_override
+        wait_done_loading r
+    }
+    proc create_emptyset {key members} {
+        r del $key
+        foreach member $members { r sadd $key $member }
```

Comment:
I don't personally like that in order to create an empty set, you first need to add an arbitrary element then delete that element. The primary use case for this feature seems to be for query caching, where really your goal is to empty the set (or create a new empty set)

I left a comment in the issue proposing a new API - `SEMPTY` or something similar (maybe `SCREATE`). Even if we decide on a different approach than what I was suggesting there - I think an API like this will be needed to make it user friendly